### PR TITLE
sprint01/created-basic-apis-for-practice

### PR DESCRIPTION
**Pull request by AfaqAhmed-XGrid**
Addresses #2

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+const express = require('express');
+const app = express();
+const path = require('path');
+
+app.use(express.json())
+
+app.get('/', (req, res) => {
+    res.status(200);
+    res.json({message: 'Welcome to the root URL of the server'})
```
Comment (junaidtariqxgrid) [inline index.js]: Please add strings to constants

File: index.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+})
+
+app.get('/hello', (req, res) => {
```
Comment (junaidtariqxgrid) [inline index.js]: Please add these single quoted strings to constants

File: package.json [hunk 3]
```diff
@@ -6,15 +6,10 @@
   "scripts": {
```
Comment (junaidtariqxgrid) [inline package.json:6]: All URLs and strings should be added to constants.js

**Review by junaidtariqxgrid — CHANGES_REQUESTED** [review]

File: index.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
+const express = require('express');
+const app = express();
+const path = require('path');
```
Comment (FahadJalal04) [inline index.js]: Categories import statements and add a comment.
e.g. 
// Module Imports
All imports related to npm modules

// Component Imports
....

File: index.js (hunk 1 above)
Comment (FahadJalal04) [inline index.js]: Move your controller functions to a separate file and add middleware for authentication and authorization.

File: index.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+app.use(express.json())
+
+app.get('/', (req, res) => {
+    res.status(200);
+    res.json({message: 'Welcome to the root URL of the server'})
+})
+
+app.get('/hello', (req, res) => {
+    res.set('Content-Type', 'text/html')
+    res.send('<h1>Welcome to the hello URL of the server</h1>')
```
Comment (FahadJalal04) [inline index.js]: The response should be a JSON object like: 
{
   success: 0/1,
   message: 'your message',
   data: {}
}

**Review by FahadJalal04 — CHANGES_REQUESTED** [review]

File: index.js (hunk 1 above)
Comment (FahadJalal04) [reply index.js]: Not addressed

File: index.js (hunk 2 above)
Comment (FahadJalal04) [reply index.js]: Same

File: modules/practice/practice.controller.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
+//constant imports
+const constants = require("./practice.constant");
+
+const printMessage = (req, res) => {
```
Comment (FahadJalal04) [inline modules/practice/practice.controller.js]: This function should be an async function

File: modules/practice/practice.controller.js [hunk 7]
```diff
@@ -0,0 +1,19 @@
+//constant imports
+const constants = require("./practice.constant");
+
```
Comment (FahadJalal04) [inline modules/practice/practice.controller.js:3]: Add function level comments

File: modules/practice/practice.controller.js [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+const printMessage = (req, res) => {
+    res.status(200);
```
Comment (FahadJalal04) [inline modules/practice/practice.controller.js]: Use 2 spaces for indent

File: modules/practice/practice.controller.js [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+    res.json({sucess: 1, message: constants.WELCOME_MSG, data:{}})
+}
+
+const createHTML = (req, res) => {
+    res.set('Content-Type', 'text/html')
+    res.json({sucess: 1, message: constants.WELCOME_TAG, data:{}})
+}
+
+const printName = (req, res) => {
+    const {name} = req.body
```
Comment (FahadJalal04) [inline modules/practice/practice.controller.js]: No need to extract the name from the body. You can directly use req.body.name

File: modules/practice/practice.route.js [hunk 10]
```diff
@@ -0,0 +1,19 @@
+//Package Imports
+const express  =require("express");
```
Comment (FahadJalal04) [inline modules/practice/practice.route.js]: Why 2 spaces before the equal sign and no space after equal?

File: modules/practice/practice.route.js [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+const { printMessage, createHTML, printName } = require("./practice.controller");
+
+//Creating instances
+const router = express.Router();
+
+router.get('/', printMessage)
+
+router.get('/hello', createHTML)
+
+router.post('/api/post', printName)
```
Comment (FahadJalal04) [inline modules/practice/practice.route.js]: Use a separate file for routes as discussed

**Review by FahadJalal04 — CHANGES_REQUESTED** [review]

File: modules/practice/practice.controller.js (hunk 6 above)
Comment (AfaqAhmed-XGrid) [reply modules/practice/practice.controller.js]: I did not use await keywork so why to make it an async function?

File: modules/practice/practice.route.js (hunk 11 above)
Comment (AfaqAhmed-XGrid) [reply modules/practice/practice.route.js]: I have created all routes in a separate practice.route.js file

File: modules/practice/practice.controller.js (hunk 6 above)
Comment (FahadJalal04) [reply modules/practice/practice.controller.js]: Commented for future ref.

**Review by FahadJalal04 — COMMENTED** [review]
Replied

File: index.js (hunk 2 above)
Comment (AfaqAhmed-XGrid) [reply index.js]: Done!

File: package.json (hunk 3 above)
Comment (AfaqAhmed-XGrid) [reply package.json:6]: Done!

File: index.js (hunk 4 above)
Comment (AfaqAhmed-XGrid) [reply index.js]: Done!

File: index.js (hunk 1 above)
Comment (AfaqAhmed-XGrid) [reply index.js]: Done!

File: index.js (hunk 5 above)
Comment (AfaqAhmed-XGrid) [reply index.js]: Done!

File: modules/practice/practice.controller.js (hunk 6 above)
Comment (AfaqAhmed-XGrid) [reply modules/practice/practice.controller.js]: OKay!

File: modules/practice/practice.controller.js (hunk 8 above)
Comment (AfaqAhmed-XGrid) [reply modules/practice/practice.controller.js]: Done!

File: modules/practice/practice.controller.js (hunk 9 above)
Comment (AfaqAhmed-XGrid) [reply modules/practice/practice.controller.js]: Done!

File: modules/practice/practice.route.js (hunk 10 above)
Comment (AfaqAhmed-XGrid) [reply modules/practice/practice.route.js]: Done!

**Review by FahadJalal04 — APPROVED** [review]

File: modules/practice/practice.controller.js (hunk 7 above)
Comment (AfaqAhmed-XGrid) [reply modules/practice/practice.controller.js:3]: Done!

**Review by junaidtariqxgrid — APPROVED** [review]

**Review by junaidtariqxgrid — APPROVED** [review]

**Review by junaidtariqxgrid — APPROVED** [review]

**Review by junaidtariqxgrid — APPROVED** [review]

**Pull request merged**